scalameta/contrib/src/main/scala/scala/meta/contrib/DocToken.scala

```diff
@@ -0,0 +1,246 @@
+package scala.meta.contrib
+
+import scala.meta.contrib.DocToken.DocKind
+
+/**
+  * Represents a scaladoc line.
+  */
+case class DocToken(kind: DocKind, name: Option[String], body: String) {
```

Comment:
I think this ast design could improve. For example, what if there comes a new annotation with 3 arguments? Then this top-definition would need to change.

Here's an alternative
```scala
sealed abstract class DocToken
object DocToken {
  case class Param(name: String, description: String) extends DocToken
  ...
}
```

Fastparse should make it easier to handle multi-line descriptions/bodies, avoiding the need for merging lines and the `copy` macro.